core/xml.js

```diff
@@ -596,9 +601,15 @@ Blockly.Xml.domToVariables = function(xmlVariables, workspace) {
     }
     var type = xmlChild.getAttribute('type');
     var id = xmlChild.getAttribute('id');
-    var name = xmlChild.textContent;
-
-    workspace.createVariable(name, type, id);
+    var name = '';
+    for (var i = 0; i < xmlChild.childNodes.length; i++) {
+      if (xmlChild.childNodes[i].nodeType === Node.TEXT_NODE) {
```

Comment:
Use ``Blockly.utils.dom.Node.TEXT_NODE`` instead for Node.js compatibility. Why not check for ``childNodes[i].nodeName`` == 'data'`` here and populate your data attribute.

Since we're only writing a single name / data attribute, I think it's fair to override rather than append when populating each attribute.

Reply:
Okay, changed. Variable node can have more textNodes when is XML prettified (spaces before and after data node). So I have to append name in my opinion.